### DISCUSSION TITLE
Stuck force closure TX, unable to bumpfee

### DISCUSSION THREAD

**Discussion by feelus** [lightningnetwork/lnd#7779] [Troubleshooting] [answered]
Hello,

My CLN node (node A, therefore no anchors) had a public channel with a node running LND (node B). We ran into an HTLC timeout and node A (CLN) initiated a force closure. Since the channel was fairly old, the [close/scratch TX](https://blockchair.com/bitcoin/transaction/d961b2abed1b5a7427676d5b4a2dca02dcd11c3d5d3dd35a0e0a566eb9527947) had lower fees and is sitting in the mempool for more than 3 weeks now. 

Since the TX wasn't getting confirmed node B (LND) tried force closing the channel with it's own TX

`lncli closechannel --force e020bb832e88d2e64925284c35123c4fbe8c4e1bd0cac2f23c8e45de2ac1708e 1`

but it didn't publish any FC TX and the channel became borked and has flag `ChanStatusLocalCloseInitiator`

```
lncli pendingchannels
{
    "total_limbo_balance": "50949",
    "pending_open_channels": [
    ],
    "pending_closing_channels": [
    ],
    "pending_force_closing_channels": [
    ],
    "waiting_close_channels": [
        {
            "channel": {
                "remote_node_pub": "03851aa559c69fc719f84df6c480bf75654561653c8750d260703fc24f2e078168",
                "channel_point": "e020bb832e88d2e64925284c35123c4fbe8c4e1bd0cac2f23c8e45de2ac1708e:1",
                "capacity": "5000000",
                "local_balance": "50949",
                "remote_balance": "4912754",
                "local_chan_reserve_sat": "50000",
                "remote_chan_reserve_sat": "50000",
                "initiator": "INITIATOR_LOCAL",
                "commitment_type": "STATIC_REMOTE_KEY",
                "num_forwarding_packages": "0",
                "chan_status_flags": "ChanStatusBorked|ChanStatusCommitBroadcasted|ChanStatusLocalCloseInitiator",
                "private": false
            },
            "limbo_balance": "50949",
            "commitments": {
                "local_txid": "94ae314a4dde652469b6e598031d33f03569353daf92a30003459c46be5c7a76",
                "remote_txid": "d961b2abed1b5a7427676d5b4a2dca02dcd11c3d5d3dd35a0e0a566eb9527947",
                "remote_pending_txid": "cf923eae6fc48f17f64d583ec31107946c58e96764ae5daeaf934c46c6bc83a7",
                "local_commit_fee_sat": "1718",
                "remote_commit_fee_sat": "1235",
                "remote_pending_commit_fee_sat": "0"
            },
            "closing_txid": "94ae314a4dde652469b6e598031d33f03569353daf92a30003459c46be5c7a76"
        }
    ]
}
```

Now when node B wants to `bumpfee` he's unable to do it

```
lncli wallet bumpfee --sat_per_byte 40 d961b2abed1b5a7427676d5b4a2dca02dcd11c3d5d3dd35a0e0a566eb9527947:0
[lncli] rpc error: code = Unknown desc = the passed output does not belong to the wallet
```

The problem as we see it is that node B (LND) when issuing it's force close, after node A (CLN) has already done it and it propagated through the mempool, the force closure TX from node B didnt propagate but it did get into node B mempool, replacing the original close TX. 

So node B (LND) cannot bumpfee node A's close TX, because it's not in it's mempool as both TXs spend the same inputs and node A (CLN) cannot see node B's close TX either.

This is the output from node B (LND) when checking node A's close TX (the first one that propagated)
```
$ bitcoin-cli getrawtransaction d961b2abed1b5a7427676d5b4a2dca02dcd11c3d5d3dd35a0e0a566eb9527947
error code: -5
error message:
No such mempool or blockchain transaction. Use gettransaction for wallet transactions.
```

This is the output from node B (LND) when checking it's own close TX (the one that won't propagate)
```
$ bitcoin-cli getrawtransaction 94ae314a4dde652469b6e598031d33f03569353daf92a30003459c46be5c7a76
020000000001018e70c12ade458e3cf2c2cad01b4e8cbe4f3c12354c282549e6d2882e83bb20e0010000000008c41e80031287000000000000220020d4d33e583ef265ef5ac0a3169a88e29fc7401b0274401a5bfcec7b5578f65ce905c7000000000000220020d467929215322941b1f6513e660780ea7313cd70bbe1e37c531b3e7c2860b0e772f64a000000000016001412b0baf510f9c79f996757610c9fe85845a118c604004730440220663631ebea8ef5b9489ce9b330b56b39ada38a0cbc757679d606bf511a163fe402200b93d78bb0f0c6aae128b85a71cc8f2d511801d7f68f81761de8aaab413a07590148304502210091412c47338efc04378d2d105d27e00df00c43cee125fb4256cbdba4612ebdc802204b03c0e71c787b23bdb0b152afea4916bc926b14488f8aeedab48578f31831730147522102b22ac86cf8decacbee4870f8711c12650fea98d48e3d6feb40f66b09681c96462102fdb633e2469112e624aff759cb53b0a180aafd69b8ed5c61812efdee7ef0b8bf52ae76d40520
```

**Now we are stuck - neither node can CPFP, is there anything that can be done so node B (LND) forgets it's own close TX so that it can CPFP the original close TX?**

Thanks in advance!

**Comment by paulofelipe84** [comment]
Hi, "Node B" here. Happy to answer any questions and try any suggestions.

**Comment by GeorgeTsagk** [comment, marked as answer]
Hello

Node B (LND) could try to use [chantools](https://github.com/lightninglabs/chantools) in order to bump the closing tx. Specifically you should look into the command `sweepremoteclosed`. This will use a public API (default is blockstream.info/api) so the CLN tx should be visible there.

**Reply by paulofelipe84** [reply]
Hey, I did it:

`chantools sweepremoteclosed --feerate 40 --sweepaddr bc1q35rup42t6w3hgtwt93nt3v8nd4730f6r7gslft`

It did find unspent outputs

`2023-06-22 09:53:58.696 [INF] CHAN: Found 1 unspent outputs for address bc1q35rup42t6w3hgtwt93nt3v8nd4730f6r7gslft`

It generated the transaction, but:

`error code: -25
error message:
bad-txns-inputs-missingorspent`

We're thinking this might be happening because of my FC tx that's sitting only on my own mempool.

**Reply by guggero** [reply]
You need to specify an address from your wallet for `--sweepaddr`, that's where the funds will be swept to.
Hmm, the `lncli pendingchannels` output shown above is from your node, right? Then this command should have worked. Can you post the full transaction that was generated but not accepted please?

**Reply by feelus** [reply]
Thank you guys for help, it worked! 

Paulo tried to broadcast the raw TX given by `sweepremoteclosed` manually using `bitcoin-cli sendrawtransaction`, which gave the error above `bad-txns-inputs-missingorspent`. Maybe it would have worked had we tried `sweepremoteclosed` with the `--publish` flag. Anyways we took the raw TX and broadcasted it using my bitcoin node, which accepted it no problem and now it's confirmed and the channel is closed. Thanks again, happy it worked!
